initialize frontend directory

nadmax opened the pull request:
## What's new?
**Frontend**
- **Framework & Tooling**:  
  - React + TypeScript for type-safe, component-driven development  
  - Vite for fast bundling and development server  
  - TailwindCSS + PostCSS for utility-first styling  
  - ESLint and TypeScript configs for code quality and type safety  

- **Structure**:
  - `src/` with clear separation of:
    - `components/` – reusable UI components (e.g., `ConstellationView`, `ControlPanel`, `ResourcePanel`)  
    - `pages/` – main views (`Landing`, `Visualization`)  
    - `services/` – API calls and integration with backend  
    - `utils/` – helpers such as `graphProcessor`  
    - `types/` – TypeScript interfaces for API and app state  
  - `public/` for static assets  

- **Testing**:
  - Component and utility tests (`*.test.tsx`, `*.test.ts`)  
  - Test setup configured in `src/test/setup.ts`  

- **Docker**:
  - Dockerfile for containerized frontend builds  

**Infrastructure**
- Updated `compose.yml` to include the `frontend` service, building its Docker image alongside the backend  

**CI/CD**
- Add build and test jobs in parallel for the `frontend` and `backend` services  

**Other**
- Base files: `index.html`, `main.tsx`, global CSS, and configs (`tsconfig`, `vite.config.ts`, `tailwind.config.js`)

nadmax committed:
feat(frontend): add landing page

nadmax committed:
fix(frontend): center navigation tabs

nadmax committed:
chore(compose): add frontend service in compose.yml

nadmax committed:
chore(ci): use matrix strategy for backend and frontend builds and tests

nadmax committed:
feat(frontend): integrate backend API for tfstate parsing

nadmax committed:
feat(frontend): add Terraform state visualizer

nadmax committed:
feat(frontend): increase constellation opacity

nadmax committed:
fix(frontend): resolve node selection in 3D constellation view after camera movement

- Replace raycasting with screen-space distance checking for more reliable node selection
- Calculate dynamic click threshold based on sphere size and camera distance
- Force update camera and mesh matrices before projection to handle OrbitControls transformations
- Increase sphere sizes from 12/8 to 20/15 for better visibility and clickability
- Set click threshold to 1.0 normalized units for consistent selection across all camera positions
- Fix dragging detection using pointer movement distance instead of OrbitControls events

This fixes the issue where clicking on nodes would not open the ResourcePanel, especially after rotating or zooming the constellation view

nadmax committed:
chore(Dockerfile): remove packages versions for the frontend's Dockerfile to avoid image build error

sofianedjerbi commented on frontend/src/components/ConstellationView.tsx:
```diff
@@ -0,0 +1,398 @@
+        };
+        window.addEventListener('resize', handleResize);
+
+        const animate = () => {
```
Pense a call pour eviter les memleak: https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame a la fin

sofianedjerbi commented on frontend/Dockerfile:
```diff
@@ -0,0 +1,24 @@
+COPY package*.json ./
+RUN npm ci --omit=dev
+
+FROM alpine:3.22 AS final
```
pourquoi pas node direct ? node:22-alpine

nadmax replied:
L'image a été remplacé par `caddy:2-alpine`

sofianedjerbi reviewed (changes requested):
C'est cool globalement juste les 2 memleak à confirmer

nadmax committed:
chore(docker): reduce frontend's docker image by using caddy:2-alpine instead of alpine:3.22

nadmax committed:
fix(frontend): prevent memory leaks in ConstellationView Three.js component

- Store and cancel requestAnimationFrame ID in cleanup to prevent animation loop leak
- Dispose Three.js geometries and materials when removing objects from scene
- Properly clean up node objects, edge objects, and their child meshes (glow effects)
- Add animationFrameIdRef to track animation frame for cancellation

These changes prevent GPU and CPU memory accumulation when filters change or component unmounts.

nadmax commented on frontend/src/components/ConstellationView.tsx:
```diff
@@ -0,0 +1,398 @@
+                material.emissiveIntensity = 0.5;
+            }
+        });
+    }, [selectedNodeId]);
```
Au temps pour moi, il faut ajouter graph et filters en dépendances pour que l'effet s'active aussi quand le graph change et quand les filtres changent   
Je vais commit ça

nadmax committed:
fix(frontend): add graph and filters dependencies to node emission effect

Add missing dependencies (graph, filters) to useEffect that updates node
emissive properties. This ensures materials are properly recalculated when
the graph structure or filter state changes, not only when selectedNodeId
changes.

nadmax committed:
fix(frontend): add VITE_ prefix to API_URL environment variable

Update API_BASE_URL to use VITE_API_URL instead of API_URL to properly
expose the environment variable to the client-side code in Vite.

nadmax committed:
perf(frontend): reduce unnecessary ControlPanel re-renders and iterations

nadmax committed:
feat(frontend): add environment variable support for domain configuration

Replace hardcoded :3000 with {:localhost:3000} to enable
automatic TLS certificate provisioning via Let's Encrypt when deployed
with a real domain name. Falls back to localhost:3000 for local development.

nadmax committed:
feat(CI): add lint job before build

Also fix the errcheck linting in
backend/internal/handlers/parse.go:26:28
